WordPress/Classes/ViewRelated/Blog/SharingAuthorizationWebViewController.m

```diff
@@ -0,0 +1,160 @@
+#import "SharingAuthorizationWebViewController.h"
+#import "Blog.h"
+#import "Publicizer.h"
+#import "WordPressAppDelegate.h"
+#import "WPUserAgent.h"
+
+#pragma mark - SharingAuthorizationWebViewController
+
+/**
+ *	@brief	classify actions taken by web API
+ */
+typedef enum {
+    AuthorizeActionNone,
+    AuthorizeActionUnknown,
+    AuthorizeActionRequest,
+    AuthorizeActionVerify,
+    AuthorizeActionDeny,
+} AuthorizeAction;
+
+
+@interface SharingAuthorizationWebViewController ()
+
+/**
+ *	@brief	verification loading -- dismiss on completion
+ */
+@property (nonatomic, assign) BOOL loadingVerify;
+/**
+ *	@brief	blog a publicizer is being authorized for
+ */
+@property (nonatomic, strong) Blog *blog;
+/**
+ *	@brief	publicize service being authorized
+ */
+@property (nonatomic, strong) Publicizer *publicizer;
+
+@end
+
+
+@implementation SharingAuthorizationWebViewController
+
++ (instancetype)controllerWithPublicizer:(Publicizer *)publicizer
+                                 forBlog:(Blog *)blog
+{
+    NSParameterAssert(publicizer);
+    NSParameterAssert(blog);
+    
+    // some services require Safari user agent to log in
+    [[WordPressAppDelegate sharedInstance].userAgent useDefaultUserAgent];
```

Comment:
Safer to put it in `viewDidLoad` before superclass kicks off the first request I think.